hare/builder.go

```diff
@@ -41,120 +43,132 @@ func (m *Message) Field() log.Field {
 	return log.String("message", m.String())
 }
 
-// certificate is a collection of messages and the set of values.
+// Certificate is a collection of messages and the set of values.
 // Typically used as a collection of commit messages.
-type certificate struct {
+type Certificate struct {
 	Values  []types.ProposalID // the committed set S
-	AggMsgs *aggregatedMessages
+	AggMsgs *AggregatedMessages
 }
 
-// aggregatedMessages is a collection of messages.
-type aggregatedMessages struct {
-	Messages []*Message
+// AggregatedMessages is a collection of messages.
+type AggregatedMessages struct {
+	Messages []Message
 }
 
-// innerMessage is the actual set of fields that describe a message in the Hare protocol.
-type innerMessage struct {
-	Type             messageType
+// InnerMessage is the actual set of fields that describe a message in the Hare protocol.
+type InnerMessage struct {
+	Type             MessageType
 	InstanceID       types.LayerID
 	K                uint32 // the round counter
 	Ki               uint32
 	Values           []types.ProposalID  // the set S. optional for commit InnerMsg in a certificate
 	RoleProof        []byte              // role is implicit by InnerMsg type, this is the proof
 	EligibilityCount uint16              // the number of claimed eligibilities
-	Svp              *aggregatedMessages // optional. only for proposal Messages
-	Cert             *certificate        // optional
+	Svp              *AggregatedMessages // optional. only for proposal Messages
+	Cert             *Certificate        // optional
 }
 
 // Bytes returns the message as bytes.
-func (im *innerMessage) Bytes() []byte {
+func (im *InnerMessage) Bytes() []byte {
 	buf, err := codec.Encode(im)
 	if err != nil {
 		log.Panic("could not marshal InnerMsg before send")
 	}
 	return buf
 }
 
-func (im *innerMessage) String() string {
+func (im *InnerMessage) String() string {
+	if im == nil {
+		return ""
+	}
 	return fmt.Sprintf("Type: %v InstanceID: %v K: %v Ki: %v", im.Type, im.InstanceID, im.K, im.Ki)
 }
 
-// messageBuilder is the impl of the builder DP.
+// MessageBuilder is the impl of the builder DP.
 // It allows the user to set the different fields of the builder and eventually Build the message.
-type messageBuilder struct {
+type MessageBuilder struct {
 	msg   *Msg
-	inner *innerMessage
+	inner *InnerMessage
 }
 
 // newMessageBuilder returns a new, empty message builder.
 // One should not assume any values are pre-set.
-func newMessageBuilder() *messageBuilder {
-	m := &messageBuilder{&Msg{Message: &Message{}, PubKey: nil}, &innerMessage{}}
+func newMessageBuilder() *MessageBuilder {
```

Comment:
you definitely shouldn't generate scale codec for MessageBuilder

Reply:
done, cheers